yt_dlp/extractor/thirtydaysinger.py

```diff
@@ -0,0 +1,107 @@
+import re
+
+from .wistia import WistiaBaseIE
+from ..utils import (
+    clean_html,
+    get_elements_html_by_class
+)
+
+
+class ThirtyDaySingerBase(WistiaBaseIE):
+    _INDEX_EXTRACTION_RE = r'/tutorial/[\w-]+/(?P<index>[\w-]+)'
+
+    def _extract_for_url(self, url):
+        lesson_index = re.search(self._INDEX_EXTRACTION_RE, url).group('index')
+        webpage = self._download_webpage(url, lesson_index)
+        match = next(self._extract_wistia_async_embed(webpage))
+        embed_config = self._download_embed_config('medias', match.group('id'), url)
+
+        embed_infojson = self._extract_media(embed_config)
+        webpage_infojson = self._extract_webpage_data(webpage)
+
+        return {**embed_infojson, **webpage_infojson}
+
+    def _extract_webpage_data(self, webpage):
+        title = self._html_search_regex(r'<h1>([^<]+)</h1>', webpage, 'title')
+        fallback_title = self._html_extract_title(webpage)
+        description = self._html_search_meta('description', webpage, fatal=False)
+
+        return {
+            'title': title or fallback_title,
+            'description': clean_html(self._format_html_list(description))
+        }
+
+    # The site makes extensive use of HTML lists for formatting and `clean_html`
+    # doesn't handle them well. This is needed to keep lists readable.
+    def _format_html_list(self, html):
+        replacements = {
+            '<ul>': '<br>',
+            '</ul': '<br>',
+            '<li>': '<br>- ',
+            '</li>': ''
+        }
+
+        for k, v in replacements.items():
+            html = html.replace(k, v)
+
+        return html
+
+
+class ThirtyDaySingerIE(ThirtyDaySingerBase):
+    _VALID_URL = r'https?://www\.30daysinger\.com/tutorial/[\w-]+/[\w-]+'
```

Comment:
For viz: I'm not creating a match group for the video index here since that's handled in `ThirtyDaySingerBase`. Doing it this way allows me to share the logic between the video and playlist extractor

Reply:
I'm not a fan of doing it this way tbh. In the baseclass you could have the full url pattern as `_URL_REGEX` and then in the subclass do `_VALID_URL = ThirtyDaySingerBaseIE._URL_REGEX`. Then you can use `_match_id` or `_match_valid_url` in the baseclass method